cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -592,7 +592,6 @@ private ArtifactClasses inspectArtifact(Id.Artifact artifactId, File artifactFil
     } else {
       return ArtifactClasses.builder()
         .addApps(artifactClasses.getApps())
-        .addDatasets(artifactClasses.getDatasets())
```

Comment:
It does. With the new ClassLoader, actually we don't need to grab the dataset classes during artifact deployment. Whether a class is a dataset class is determined dynamically during classloading.